ci-operator/step-registry/cucushift/installer/check/nutanix/zones/cucushift-installer-check-nutanix-zones-commands.sh

```diff
@@ -24,22 +23,46 @@ if [[ "$SINGLE_ZONE" == "true" ]]; then
         fi
     done
 else
-    echo "Check node running on multy zone: $prism_element1"
-    # shellcheck disable=SC2076
-    if ! [[ " ${node_zone_list[*]} " =~ " $prism_element1 " ]]; then
-        echo "Fail: fail to check zone: $prism_element1 exist"
-        exit 1
+    default_node_pes=$(echo "$prism_element1" "$prism_element2" "$prism_element3" | tr " " "\n" | sort -u | xargs)
+    control_plane_pes="$default_node_pes"
+    compute_pes="$default_node_pes"
+    compute_zone_list=$(oc get nodes -l node-role.kubernetes.io/worker= -o=jsonpath='{.items[*].metadata.labels.topology\.kubernetes\.io\/zone}' | tr " " "\n" | sort -u | xargs)
+    control_plane_zone_list=$(oc get nodes -l node-role.kubernetes.io/master= -o=jsonpath='{.items[*].metadata.labels.topology\.kubernetes\.io\/zone}' | tr " " "\n" | sort -u | xargs)
+
+    if [[ "$CONTROL_PLANE_ZONE" != "" ]]; then
+        control_plane_pes=$(echo "$COMPUTE_ZONE" | sed -e "s/failure-domain-1/$prism_element1/g" -e "s/failure-domain-2/$prism_element2/g" -e "s/failure-domain-3/$prism_element3/g" | tr " " "\n" | sort -u | xargs)
     fi
-    echo "Check node running on multy zone: $prism_element2"
-    # shellcheck disable=SC2076
-    if ! [[ " ${node_zone_list[*]} " =~ " $prism_element2 " ]]; then
-        echo "Fail: fail to check zone: $prism_element2 exist"
-        exit 1
+    if [[ "$COMPUTE_ZONE" != "" ]]; then
+        compute_pes=$(echo "$COMPUTE_ZONE" | sed -e "s/failure-domain-1/$prism_element1/g" -e "s/failure-domain-2/$prism_element2/g" -e "s/failure-domain-3/$prism_element3/g" | tr " " "\n" | sort -u | xargs)
     fi
-    echo "Check node running on multy zone: $prism_element3"
-    # shellcheck disable=SC2076
-    if ! [[ " ${node_zone_list[*]} " =~ " $prism_element3 " ]]; then
-        echo "Fail: fail to check zone: $prism_element3 exist"
+    if [[ $control_plane_zone_list == "$control_plane_pes" ]]; then
+        echo "Pass: passed to check control plane zone: $control_plane_zone_list, expected: $control_plane_pes"
+    else
+        echo "Fail: fail to check control plane zone: $control_plane_zone_list, expected: $control_plane_pes"
         exit 1
     fi
+    IFS=" " read -r -a array_compute_pes <<<"$compute_pes"
+    IFS=" " read -r -a array_compute_zone_list <<<"$compute_zone_list"
+    if [[ "$COMPUTE_REPLICAS" != "" ]] && [[ $COMPUTE_REPLICAS < ${#array_compute_pes[@]} ]]; then
+        # When compute replicas < zone
+        # Check compute zone list number equals compute replicas num
+        if [[ ${#array_compute_zone_list[@]} == "$COMPUTE_REPLICAS" ]]; then
```

Comment:
I do not think we need this condition, once COMPUTE_REPLICAS less than failure domains, then iterate compute_zone_list, then ensure each element is in compute_pes.

Reply:
yes, to avoid bug like all the computes are landed to "$prism_element1"